### PR TITLE
issue 821 fix proposal

### PR DESCRIPTION
**Pull request by gwenaelp**
make reporter default option stop overriding reporter option in configuration files, to allow defining custom reporters again (issue #821)

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ function App(config, finalizer) {
 
   this.reportFileName = this.config.get('report_file');
   this.reportFileStream = this.initReportFileStream(this.reportFileName);
-  this.reporter = this.initReporter(this.config.get('reporter'), this.reportFileStream);
+  this.reporter = this.initReporter((this.config.get('reporter') || 'tap'), this.reportFileStream);
```
Comment (johanneswuerbach) [inline lib/app.js]: This should be required as `tap` is still the default https://github.com/testem/testem/blob/master/lib/config.js#L160.

**Comment by johanneswuerbach** [comment]
Could you add a test so this won't break again? :-)

**Comment by DerivcoHK** [comment]
Can you please merge this fix as soon as possible ?

**Comment by johanneswuerbach** [comment]
Not without tests to prevent further regressions. My time is a bit limited atm, but PRs are welcome.

**Comment by DerivcoHK** [comment]
**config.js**
```
Config.prototype.defaults = {
  ....
  reporter: 'tap',
   ....
};
```

```
Config.prototype.getConfigProperty = function (key) {
  if (this.config && key in this.config) {
    return this.config[key];
  }
  if (key in this.progOptions && typeof this.progOptions[key] !== 'undefined') {
    return this.progOptions[key];
  }
  if (key in this.fileOptions && typeof this.fileOptions[key] !== 'undefined') {
    return this.fileOptions[key];
  }
  if (key in this.defaults) {
     ....
      return defaultVal;
  }
}
```

**testem.js**
`  .option('-R, --reporter [reporter]', 'Test reporter to use [tap|dot|xunit]', 'tap')
`

Now the priority of progOptions is higher than that of fileOptions, and we have default progOptions for reporter 'tap' so the "reporter" option in config cannot be used anymore. 

It is safe to remove the default value 'tap' in the -R option, because it will get the default value when there are no progOptions and fileOptions.

How can I add a test to show it is safe and so that you can merge it?

**Comment by johanneswuerbach** [comment]
Make sense, could you send a PR just removing the `tap`?

**Comment by johanneswuerbach** [comment]
Fixed by https://github.com/testem/testem/pull/860

**Pull request closed without merging**